stew/ptrops.nim

```diff
@@ -36,9 +36,12 @@ template offset*[T](p: ptr T, count: int): ptr T =
   # better optimizations.
 
   # We turn off checking here - too large counts is UB
-  {.checks: off.}
-  let bytes = count * sizeof(T)
-  cast[ptr T](offset(cast[pointer](p), bytes))
+  {.push checks: off.}
```

Comment:
the other option is that we cast things to `uint` instead of messing with checks

Reply:
That would also avoid the issue it's triggering in 1.2 with some macro expansion error message in CI.

Reply:
Ah, see elsewhere, those CI failures were already there.

Reply:
For the purposes of this PR, I think I want to keep it very minimal -- removing `checks` by using `uint` is laudable, but slightly riskier. This is a fix we can deploy immediately, because it just matches the actual execution model to the intended execution model on the callers' side.